src/Jab/ContainerGenerator.cs

```diff
@@ -358,7 +362,7 @@ private void WriteServiceProvider(CodeWriter codeWriter, ServiceProvider root)
             {
                 if (rootRootCallSite.IsMainImplementation)
                 {
-                    codeWriter.Append($"if (type == typeof({rootRootCallSite.ServiceType})) return ");
+                    codeWriter.Append($"if (type == typeof({rootRootCallSite.ServiceType.ToDisplayString(NullableFlowState.NotNull)})) return ");
```

Comment:
Do we ever render nullable annotation? If not, can we change https://github.com/pakrym/jab/blob/main/src/Jab/CodeWriter.cs#L173 to always strip it?

Reply:
Even better is to strip the nullability from ServiceType when building the service tree and callsites. That way, callsites are correct in what they represent. Logically, the service type and implementation type can't be nullable.

Reply:
I didn't looked to much, but I can't create a ITypeSymbol because all implementations are internal. Maybe can create a wrapper.

Reply:
Works for me!